### PR TITLE
[batch] Miscellaneous performance improvements

### PR DESCRIPTION
**Pull request by jigold**
- renamed `get_record` to `get_records` to be clear that this returns all matching records
- Added `state` and `complete` to the Batch class in server. These are computed when the Batch is queried from the database.
- Batch `to_dict` returns `state` and `complete` in the result. It also has an option on whether to include `jobs` in the output.
- Batch `list_batches` in `server.py` no longer outputs the `jobs` per Batch
- Changed `ci2` to use the new Batch status with the state and complete included.

### CODE REVIEW

**Comment by jigold** [comment]
@cseed  I think these are some of the changes we discussed today, but I'm not sure if I actually made things better by adding the more complicated query for getting batch records. Can you quickly look over?

File: batch/batch/server/server.py [hunk 1]
```diff
@@ -785,18 +799,22 @@ def from_record(record):
...
         self.callback = callback
         self.ttl = ttl
         self.is_open = is_open
         self.userdata = userdata
         self.user = user
+        self.state = state
+        self.complete = complete
 
     async def get_jobs(self):
         job_ids = await db.batch_jobs.get_jobs(self.id)
```
Comment (cseed) [inline batch/batch/server/server.py]: This still needs to get fixed.  `get_jobs` should just query and return the full records.

Comment (cseed) [reply batch/batch/server/server.py]: Oops, this was in another PR, sorry.

**Comment by jigold** [comment]
Stacked on #5981

**Comment by jigold** [comment]
Closing in favor of future PR.

**Pull request closed without merging**